cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go

```diff
@@ -46,17 +41,7 @@ func main() {
 	flag.Parse()
 	glog.Info("Kubernetes Elasticsearch logging discovery")
 
-	settings, err := clientcmd.LoadFromFile(*kubeconfig)
-	if err != nil {
-		glog.Fatalf("Error loading configuration from %s: %v", *kubeconfig, err.Error())
-	}
-
-	config, err := clientcmd.NewDefaultClientConfig(*settings, &clientcmd.ConfigOverrides{}).ClientConfig()
-	if err != nil {
-		glog.Fatalf("Failed to construct config: %v", err)
-	}
-
-	c, err := client.New(config)
+	c, err := client.NewInCluster()
```

Comment:
Right, this code is designed to *only* run in a cluster. What's wrong with that? I thought this would be a not totally uncommon pattern?